src/xtgeo/surface/regular_surface.py

```diff
@@ -2043,67 +2043,75 @@ def operation(self, opname, value):
     # Operations restricted to inside/outside polygons
     # ==================================================================================
 
-    def operation_polygons(self, poly, value, opname="add", inside=True):
+    def operation_polygons(self, poly, value, opname="add", inside=True, _version=1):
```

Comment:
Is there a specific reason that the version is defaulted to 1 here and not the the new improved algorithm?

Reply:
Good catch, was initially conservative - but all tests indicate exactly same result in v1 vs v2, so I will change to 2.